playbooks/robusta_playbooks/prometheus_enrichments.py

```diff
@@ -0,0 +1,44 @@
+from robusta.api import *
+
+
+def parse_timestamp_string(date_string: str) -> datetime:
+    try:
+        if date_string:
+            return datetime.strptime(date_string, "%Y-%m-%d %H:%M:%S")
+        return None
+    except Exception:
+        logging.error(f"Failed parsing time string {date_string}", exc_info=True)
+        return None
+
+
+@action
+def prometheus_enricher(event: ExecutionBaseEvent, params: PrometheusQueryParams):
+    """
+        Enriches the finding with a prometheus query
+
+        for example prometheus queries see here:
+        https://prometheus.io/docs/prometheus/latest/querying/examples/
+    """
+    if not params.promql_query or (not params.graph_duration_minutes and not params.starts_at):
```

Comment:
I think promql_query should be mandatory in the params.

I'm not sure what is the allowed configuration options.
Is it either `graph_duration_minutes` or both `starts_at` and `ends_at`?
Is there a way to express that in the Pydantic class?

```
class PrometheusDuration(BaseModel):
    duration_minutes: int


class PrometheusDateRange(BaseModel):
    starts_at: str
    ends_at: str


class PrometheusQueryParams(PrometheusParams):
    """
    :var promql_query: the prometheusql query you want to run
    :var graph_duration_minutes: the amount of minutes back you want results for - this is not needed if starts_at is defined
    :var starts_at: the start date/time for the query - this is not needed if graph_duration_minutes is defined
    :var end_time: the end time for the query, if none is set than it defaults to now

    both starts at and ends_at use format "%Y-%m-%d %H:%M:%S"

    """
    promql_query: str
    duration: Union[PrometheusDateRange, PrometheusDuration]
```